src/ucp/core/ucp_context.h

```diff
@@ -35,12 +35,14 @@ enum {
 
 
 struct ucp_config {
-    /** Array of device names to use */
-    UCS_CONFIG_STRING_ARRAY_FIELD(names)   devices;
+    /** Array of network device names to use */
+    str_names_array_t net_devices;
```

Comment:
maybe declare an array of arrays here, so it could be accessed by device type:
```
str_names_array_t devices[UCT_DEVICE_TYPE_LAST];
```

and then:
```
check_koo(..., &config->devices[device_type])
```